include/aws/common/atomic.h

```diff
@@ -0,0 +1,68 @@
+#ifndef AWS_COMMON_ATOMIC_H
+#define AWS_COMMON_ATOMIC_H
+
+/*
+* Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#include <aws/common/common.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Atomically retrieves the value at `dst`, and returns it.
+ */
+AWS_COMMON_API static int aws_atomic_get(int *dst);
+
+/**
+ * Atomically sets the value at `dst` to `value`.
+ */
+AWS_COMMON_API static int aws_atomic_set(int *dst, int value);
+
+/**
+ * Adds `addend` to `dst` and returns the old value from `dst`.
+ */
+AWS_COMMON_API static int aws_atomic_add(int *dst, int addend);
```

Comment:
These are getting waaaaay more verbose than necessary. I will change them, but heavily favor:

`aws_atomic_cas`

since the terseness here will really be appreciated. To me atomics like these are a lot like math helper functions (3D math in games). Someone writing, for example, a physics solver will use a very terse language since they are A) extremely familiar with 3D math, and B) readability of the "macro" algorithm is way more important than expressiveness of the "micro" function name.